src/store/cpak.rs

```diff
@@ -56,12 +56,13 @@ impl Cpak {
     pub fn parse_pkey(&mut self) -> Result<(), Error> {
         let s = self.raw_pkey.get();
 
-        let pkey = serde_json::from_str::<JsonWebKey>(s)
-            .map_err(|e| Error::Syntax(e.to_string()))?
-            .clone();
+        let pkey = serde_json::from_str::<jwk::Jwk>(s).map_err(|e| Error::Syntax(e.to_string()))?;
 
         self.pkey = Some(pkey);
 
         Ok(())
     }
+    pub fn get_pkey(&mut self) -> Result<Option<jwk::Jwk>, Error> {
+        Ok(self.pkey.clone())
+    }
```

Comment:
instead of cloning we could just return a ref?

```suggestion

    pub fn get_pkey(&mut self) -> Result<Option<&jwk::Jwk>, Error> {
        Ok(self.pkey.as_ref())
    }
```

Reply:
done